### PR TITLE
e2e, SR-IOV: Ignore warnings due downward API mount not ready on time

### PR DESCRIPTION
**Pull request by ormergi**
### What this PR does
Following #12691, this PR mitigates SR-IOV lane breakage by temporarily set e2e tests ignore the mentioned warning.
In follow up PR we will introduce sustainable solution to enable having good indication in case of a real problem.

**Before this PR:**
When network-info mount is not ready on time, syncVMI rutine would raise the following error:
In a scenario virt-launcher pod network-info mount is not populated on time, syncVMI routine raises an error with no context making it impossible to trace when debugging:
```
{ Failure tests/network/sriov.go:257
Unexpected Warning event received: testvmi-p4f9s,e84120d3-398c-480f-a1bf-68a0abf6cb19: server error. command SyncVMI failed: "timed out waiting for the condition"
Expected
    <string>: Warning
not to equal
    <string>: Warning
tests/watcher/watcher.go:195}
```

**After this PR:**
Error will be more descriptive:
```
  failed to create SR-IOV hostdevices:
  failed to create PCI address pool with network status from file:
  timed out waiting for the condition:
  file is not populated with network-info
```

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer

Rebased on https://github.com/kubevirt/kubevirt/pull/12680

In follow up PR we will introduce proper solution to make tests to tolerate the warning for defined amount of occurrences. In case of a real problem we will have indication something bad happens. 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/device/hostdevice/sriov/hostdev.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func CreateHostDevices(vmi *v1.VirtualMachineInstance) ([]api.HostDevice, error)
 	netStatusPath := path.Join(downwardapi.MountPath, downwardapi.NetworkInfoVolumePath)
 	pciAddressPoolWithNetworkStatus, err := newPCIAddressPoolWithNetworkStatusFromFile(netStatusPath)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to create PCI address pool with network status from file: %w", err)
```
Comment (akalenyu) [inline pkg/virt-launcher/virtwrap/device/hostdevice/sriov/hostdev.go:56]: I would wrap it in newPCIAddressPoolWithNetworkStatusFromFile close to the polling call so it's future proof

Comment (EdDev) [reply pkg/virt-launcher/virtwrap/device/hostdevice/sriov/hostdev.go:56]: I agree, the description of the error is known to that function and not up at this level.

Also, please add a descriptive error when we return an error due to the timeout.

**Review by EdDev — CHANGES_REQUESTED** [review]

File: pkg/virt-launcher/virtwrap/device/hostdevice/sriov/hostdev.go (hunk 1 above)
Comment (ormergi) [reply pkg/virt-launcher/virtwrap/device/hostdevice/sriov/hostdev.go]: Done

**Comment by ormergi** [comment]
[changes](https://github.com/kubevirt/kubevirt/compare/7835aa3cf66ddefdd6bdd2f67fc4baf00a6872db..53e72e8a4d84e11ccc630a86da8046148f59aa21): resolve https://github.com/kubevirt/kubevirt/pull/12692#pullrequestreview-2266532135 https://github.com/kubevirt/kubevirt/pull/12692#pullrequestreview-2266701236

**Comment by ormergi** [comment]
Last commit remove the expected error to ensure its visible when SR-IOV lane fails.
Once we verify this I will remove last commit and this PR could be considered ready to merge.

/test pull-kubevirt-e2e-kind-sriov

**Comment by oshoval** [comment]
https://github.com/kubevirt/kubevirt/pull/12692/commits

commit has DEBUG word, maybe you want to remove it please, unless meant to

You might want to add please more details about the flow
that virt-handler requests too early before the file is ready hence fails, and a retry will work
and that it is not good to extend the timeout
the linked issue does already says about the downward API, but it might be better to add few words, and also on the PR itself please

**Comment by ormergi** [comment]
> https://github.com/kubevirt/kubevirt/pull/12692/commits
> commit has DEBUG word, maybe you want to remove it please, unless meant to

This is intentional to verify expected error is visible, please see above comment.

> You might want to add please more details about the flow that virt-handler requests too early before the file is ready hence fails, and a retry will work and that it is not good to extend the timeout the linked issue does already says about the downward API, but it might be better to add few words, and also on the PR itself please

I think adding text about the flow and alternatives (retries) fit the issue and not this PR, please comment about it in the issue so we wont forget.

**Comment by oshoval** [comment]
> I think adding text about the flow and alternatives (retries) fit the issue and not this PR, please comment about it in the issue so we wont forget.

i think otherwise but anyhow added here
https://github.com/kubevirt/kubevirt/issues/12691#issuecomment-2317353594

**Comment by ormergi** [comment]
[The last SR-IOV lane job failed ](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/12692/pull-kubevirt-e2e-kind-sriov/1829132452895920128)as expected and the warning error is as expected
```
{ Failure tests/network/sriov.go:416
Unexpected Warning event received: testvmi-jr8qt,9c2b09ce-5cd4-4a87-a8dd-f5539b8c1fba: server error. command SyncVMI failed: "failed to create SR-IOV hostdevices: failed to create PCI address pool with network status from file: timed out waiting for the condition: file is not populated with network-info"
Expected
    <string>: Warning
not to equal
    <string>: Warning
tests/watcher/watcher.go:195}
```

Last commit is removed.

**Review by EdDev — APPROVED** [review]
Thank you!

/approve

**Review by akalenyu — APPROVED** [review]

**Comment by acardace** [comment]
/approve
on behalf of @EdDev

**Pull request merged**